### DISCUSSION TITLE
If i want to use ALE, must am i to use lsp to?

### DISCUSSION THREAD

**Discussion by Ryangithub10** [w0rp/ale#4126] [Q&A - Ask for help with problems]
Hello, i have question beaucuse i confused.

A question was wrote in top here(a title),
But i mean if i want to use ALE, must am i to know LSP and must am i to intergrating LSP with ALE or ALE need LSP for ALE can work optimally?

I hope you understand

**Comment by hsanson** [comment]
No, ALE can use some LSP servers but it is not required to use them if you do not need them.  Check the list of supported languages and tools here: https://github.com/dense-analysis/ale/blob/master/supported-tools.md
